dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/SystemWideAlertRestRepository.java

```diff
@@ -0,0 +1,164 @@
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+import javax.servlet.ServletInputStream;
+import javax.servlet.http.HttpServletRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.alerts.AllowSessionsEnum;
+import org.dspace.alerts.SystemWideAlert;
+import org.dspace.alerts.service.SystemWideAlertService;
+import org.dspace.app.rest.exception.DSpaceBadRequestException;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.SystemWideAlertRest;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * The repository for the SystemWideAlert workload
+ */
+@Component(SystemWideAlertRest.CATEGORY + "." + SystemWideAlertRest.NAME)
+public class SystemWideAlertRestRepository extends DSpaceRestRepository<SystemWideAlertRest, Integer> {
+
+    private static final Logger log = LogManager.getLogger();
+
+    @Autowired
+    private SystemWideAlertService systemWideAlertService;
+
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    protected SystemWideAlertRest createAndReturn(Context context) throws SQLException, AuthorizeException {
+        SystemWideAlert systemWideAlert = createSystemWideAlert(context);
+        return converter.toRest(systemWideAlert, utils.obtainProjection());
+    }
+
+
+    @Override
+    @PreAuthorize("permitAll()")
+    public SystemWideAlertRest findOne(Context context, Integer id) {
+        try {
+            SystemWideAlert systemWideAlert = systemWideAlertService.find(context, id);
+            if (systemWideAlert == null) {
+                throw new ResourceNotFoundException(
+                        "systemWideAlert with id " + systemWideAlert.getID() + " was not found");
+            }
+            return converter.toRest(systemWideAlert, utils.obtainProjection());
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        return null;
+    }
+
+    @Override
+    @PreAuthorize("permitAll()")
+    public Page<SystemWideAlertRest> findAll(Context context, Pageable pageable) {
```

Comment:
Since there is only ever a *single* alert, it seems like the `findAll` should either be not implemented or Admin Only.  There's no reason to make this publicly accessible, especially since it makes it very easy to find/see "inactive" alerts.